src/parser/mage/fire/modules/features/HotStreak.js

```diff
@@ -1,139 +1,62 @@
+  onHotStreakApplied() {
+    this.totalHotStreakProcs += 1;
   }
 
-  on_toPlayer_removebuff(event) {
-    const spellId = event.ability.guid;
-    if (spellId !== SPELLS.HOT_STREAK.id && spellId !== SPELLS.KAELTHAS_ULTIMATE_ABILITY.id) {
-      return;
-    }
-    if (spellId === SPELLS.HOT_STREAK.id) {
-      this.hotStreakRemoved = event.timestamp;
-    } else if (spellId === SPELLS.KAELTHAS_ULTIMATE_ABILITY.id) {
-      this.bracerProcRemoved = event.timestamp;
-      return;
-    } else if (spellId === SPELLS.PYROCLASM_BUFF.id) {
-      this.pyroclasmProcRemoved = event.timestamp;
-      return;
-    }
-    //Check for Expired Procs, also if Fireball or Scorch was cast at the same time that Hot Streak was removed, then it was cast alongside Hot Streak (For the cast before hot streak check). Excluded Hot Streak procs during Combustion.
-    //Also checks to see if the bracer proc or pyroclasm proc was removed within 100ms of Hot Streak getting removed (i.e. they hard casted Pyroblast for the bracer/pyroclasm proc and used Hot Streak on the end of it.)
-    //Also checks to see if the player has Firestarter or Combustion and if they have Firestarter, checks to see if the boss is less than 90% health
-    if (!this.lastCastTimestamp || this.lastCastTimestamp + PROC_WINDOW_MS < this.owner.currentTimestamp) {
-      this.expiredProcs += 1;
+  onHotStreakRemoved(event) {
+    this.hotStreakRemoved = event.timestamp;
+  }
+
+  checkForExpiredProcs(event) {
+    if (!this.castTimestamp || this.castTimestamp + PROC_WINDOW_MS < event.timestamp) {
       debug && this.log("Hot Streak proc expired");
-    } else if (this.hotStreakRemoved - PROC_WINDOW_MS < this.castTimestamp || this.hotStreakRemoved - PROC_WINDOW_MS < this.bracerProcRemoved || this.hotStreakRemoved - PROC_WINDOW_MS < this.pyroclasmProcRemoved) {
-      this.castedBeforeHotStreak += 1;
-    } else if (!this.selectedCombatant.hasBuff(SPELLS.COMBUSTION.id) && (!this.selectedCombatant.hasTalent(SPELLS.FIRESTARTER_TALENT.id) || (this.currentHealth / this.maxHealth) < 0.90)) {
-      this.noCastBeforeHotStreak += 1;
-      debug && this.log("No hard cast before Hot Streak");
+      this.expiredProcs += 1;
     }
   }
 
   get usedProcs() {
-    return this.totalProcs - this.expiredProcs;
+    return this.totalHotStreakProcs - this.expiredProcs;
   }
 
   get expiredProcsPercent() {
-    return (this.expiredProcs / this.totalProcs) || 0;
+    return (this.expiredProcs / this.totalHotStreakProcs) || 0;
   }
 
   get hotStreakUtil() {
-    return 1 - (this.expiredProcs / this.totalProcs) || 0;
-  }
-
-  get castBeforeHotStreakUtil() {
-    return 1 - (this.noCastBeforeHotStreak / (this.castedBeforeHotStreak + this.noCastBeforeHotStreak));
-  }
-
-  get wastedCritsPerMinute() {
-    return this.wastedCrits / (this.owner.fightDuration / 60000);
+    return 1 - (this.expiredProcs / this.totalHotStreakProcs) || 0;
   }
 
   get hotStreakUtilizationThresholds() {
```

Comment:
I feel like these should be meaner tbh -- only time they should expire is when you are going into intermission. 95-90-85?